test/gtest/uct/test_peer_failure.cc

```diff
@@ -109,12 +112,19 @@ ucs_status_t test_uct_peer_failure::err_cb(void *arg, uct_ep_h ep,
     }
 }
 
-void test_uct_peer_failure::kill_receiver()
+void test_uct_peer_failure::kill_receiver(unsigned sender_idx)
 {
     ucs_assert(!m_receivers.empty());
-    m_entities.remove(m_receivers.front());
-    ucs_assert(m_entities.size() == m_receivers.size());
-    m_receivers.erase(m_receivers.begin());
+
+    uct_ep_h ep         = m_sender->ep(sender_idx);
+    ucs_status_t status = ep->iface->ops.ep_invalidate(ep, 0);
+    if (status == UCS_ERR_UNSUPPORTED) {
+        m_entities.remove(m_receivers.front());
+        ucs_assert(m_entities.size() == m_receivers.size());
+        m_receivers.erase(m_receivers.begin());
```

Comment:
shouldn't we use sender_idx here somehow?

Reply:
yeah, but seems like the test logic is based on this ordering, so I decided do not change this... I'll add an assert